backend/battle/forms.py

```diff
@@ -36,41 +44,35 @@ class Meta:
             "pokemon_3",
         ]
 
-    pokemon_1 = forms.IntegerField(
-        label="Pokemon 1",
-        required=True,
-        min_value=1,
-        max_value=898,
-    )
-    pokemon_2 = forms.IntegerField(
-        label="Pokemon 2",
-        required=True,
-        min_value=1,
-        max_value=898,
-    )
-    pokemon_3 = forms.IntegerField(
-        label="Pokemon 3",
-        required=True,
-        min_value=1,
-        max_value=898,
-    )
-
-    def __init__(self, *args, **kwargs):
-        super(TeamForm, self).__init__(*args, **kwargs)
-        self.fields['battle'].widget = forms.HiddenInput()
-        self.fields['trainer'].widget = forms.HiddenInput()
+    battle = forms.ModelChoiceField(queryset=Battle.objects.all())
+    trainer = forms.ModelChoiceField(queryset=User.objects.all())
```

Comment:
Those used to be hidden, any reason for not being now?

Reply:
I put this attribute directly on the template form.

Reply:
I see what you want to achieve. In this case, to simplify the code, you can use the hidden widget like this:
```python
battle = forms.ModelChoiceField(widget=forms.HiddenInput(), queryset=Battle.objects.all())
trainer = forms.ModelChoiceField(widget=forms.HiddenInput(), queryset=User.objects.all())
```

And in template, there's no need for manually using `<input>` for the hidden ones. Use instead this:
```
{% for hidden in form.hidden_fields %}
    {{ hidden }}
{% endfor %}
```

Note now you don't need anymore battle and trainer on template.

Reply:
Thank you!

Reply:
Hey, when I deleted the tags hidden on the template, I got a KeyError. It seems I did not have the battle and trainer values. Do you know what this can be?

Reply:
Did you put the `{% for hidden in form.hidden_fields %}` I said above? Note this goes inside the <form>

Reply:
You should put this AND remove the hidden tags.

Reply:
Yeah, I'm going to try again to make sure.